gitpod-io/gitpod
# [observability] Add gRPC client call and additional functional metrics

## Description
Adds client call metrics to gRPC calls form TS and Go, and makes the server -> image-builder connection use the default gRPC unary options. The lack of the latter probably was the cause of the recent image-builder related outage.

Also adds cluster registration metrics to ws-manager-bridge, and image-build counts to image-builder-mk3

Thanks @JanKoehnlein for figuring out the nitty gritty of the grpc-js interceptor business.

## Related Issue(s)
Fixes #5706, #5707

## How to test
1. start a workspace to produce some data all the way 'round
2. look at the metric endpoints of `server`, `image-builder-mk3`, `ws-manager-bridge`, `ws-manager`, `ws-proxy`, `ws-scheduler`. You should see `grpc_client_calls_total` metrics on all those services.

## Release Notes

```release-note
[server] Add gRPC client call metrics
[ws-manager-bridge] Add gRPC client call and cluster registration metrics
[image-builder-mk3] Add image-build metrics
```
